project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_visitors/c4m_og_invite_visitors.module

```diff
@@ -0,0 +1,430 @@
+}
+
+/**
+ * Returns a mail string for a variable name.
+ *
+ * Used by c4m_og_invite_visitors_mail() and the settings forms to retrieve
+ * strings.
+ */
+function _c4m_og_invite_visitors_mail_text($key, $language = NULL, $variables = array(), $replace = TRUE) {
+  $langcode = isset($language) ? $language->language : NULL;
+
+  if ($admin_setting = variable_get('c4m_og_invite_visitors_mail_' . $key, FALSE)) {
+    // An admin setting overrides the default string.
+    $text = $admin_setting;
+  }
+  else {
+    // No override, return default string.
+    switch ($key) {
+      case 'invite_group_member_subject':
+        $text = t('[personal:subject]', array(), array('langcode' => $langcode));
+        break;
+
+      case 'invite_group_member_body':
+        $text = t('<p>
+Hello,
+</p>
+ 
+<p><br>
+[user:full_name] invited you to join the following [group:node_type] at [site:name]: <strong>[group:node_title]</strong> 
+</p>
+ 
+<p>See their message below:</p>
+ 
+<p> 
+[personal:body]
+</p>
+ 
+<p>
+To accept this invitation, click <a href="[group:accept_invite_url]">Join this Group</a>.
+</p>
+ 
+<p> 
+<a href="[user:public-profile-url]">[user:full_name]</a>,
+member of [group:node_title].
+</p>', array(), array('langcode' => $langcode));
+        break;
+
+    }
+  }
+
+  if ($replace) {
+    // We do not sanitize the token replacement, since the output of this
+    // replacement is intended for an e-mail message, not a web browser.
+    return token_replace($text, $variables, array(
+      'language' => $language,
+      'callback' => 'c4m_og_invite_visitors_mail_tokens',
+      'sanitize' => FALSE,
+      'clear' => TRUE,
+    )
+    );
+  }
+
+  return $text;
+}
+
+/**
+ * Token callback to add unsafe tokens for user mails.
+ *
+ * This function is used by the token_replace() call at the end of
+ * _c4m_og_invite_visitors_mail_text() to set up some additional tokens that can
+ * be used in email messages generated by c4m_og_invite_visitors_mail().
+ *
+ * @param array $replacements
+ *   An associative array variable containing mappings from token names to
+ *   values (for use with strtr()).
+ * @param array $data
+ *   An associative array of token replacement values. If the 'user' element
+ *   exists, it must contain a user account object with the following
+ *   properties:
+ *   - login: The UNIX timestamp of the user's last login.
+ *   - pass: The hashed account login password.
+ * @param array $options
+ *   Unused parameter required by the token_replace() function.
+ */
+function c4m_og_invite_visitors_mail_tokens(&$replacements, $data, $options) {
+  global $user, $base_url;
+  if (isset($data['user'])) {
+    $group = c4m_og_current_group();
+    $url_options = array(
+      'absolute' => TRUE,
+      'purl' => array(
+        'provider' => 'og_purl|node',
+        'id' => $group->nid,
+      ),
+    );
+
+    $replacements['[user:full_name]'] = c4m_user_name($user);
```

Comment:
There is no problem with the function call here.
This function is also called by a `_user_presave` of another module (c4m_user_profile). That is what causes the error.
I meant to exit the function and used `return t('New');` instead of just `return;` because the function doc defines `@return string;` I'll change it to `return '';` to clarify the return string is meaningless.